### ISSUE TITLE
Implementing outside of oobabooga

### ISSUE THREAD

**Issue by unoriginalscreenname** [kjerk/instructblip-pipeline#1]
I was super disappointed when I tried to run blip instruct out of the box and it just didn't work.  I saw your comment on the LAVIS repo and really appreciated you trying to make this work!

I was wondering if this could be implemented or adapted outside of Oobabooga?  I wasn't able to get it running in there to test it out. 
 I was getting an error and didn't spend a lot of time digging into it to be honest.

I'm working on an image captioning project for archival photos.  I've implemented Salesforce/blip2-opt-2.7b locally and it's quite good!  However, I would love to explore the use of llms, particularly the wizard models, to help add additional user guided detail to these images.  The code to get blip2-opt running is amazingly trivial.  The oobabooga api, however, has given me some problems.  

Is it possible to implement blip-instruct outside of oobabooga without a lot of hassle?  or are you piggybacking off of their exllama and other model loading infrastructure?

**Comment by kjerk** [comment]
So I'm piggybacking off of ooba's [existing multimodal](https://github.com/oobabooga/text-generation-webui/tree/main/extensions/multimodal) pipeline implementation and GPTQ-for-llama already running on it. This way you can write an adapter (this repo) that embeds tokens and it does the rest, dealing with having a quantized model loader / inference.

It might be possible to implement this with [llama.cpp](https://github.com/ggerganov/llama.cpp/tree/b5472ea0ada081a6e1c06998ebbc9a24aa2cd4a4/examples/embd-input) but I haven't looked into it too deeply as primarily an oobabooga user.

I think huggingface just added quantized inference to transformers, so maybe it will be coming natively to transformers? (which is what salesforce implemented BLIP against)

**Comment by unoriginalscreenname** [comment]
Yeah, oobabooga does a lot of heavy lifting with all the quantized loading.  It's such a mess.  I'll look into if I can use the multimodal via the api.  I haven't dug into that at all.

**Comment by unoriginalscreenname** [comment]
They fixed this on the hugging face side now.  I was able to run it with their built in 4 or 8 bit method. 

self.model = InstructBlipForConditionalGeneration.from_pretrained(
            self.source_model,
            device_map={"":0},
            load_in_4bit=True,
            torch_dtype=torch.bfloat16
        )

Were you able to get this working with other models?  The vicuna model is unstable and I find it will get stuck in a loop of generating nonsense sometimes.

**Comment by kjerk** [comment]
> They fixed this on the hugging face side now. I was able to run it with their built in 4 or 8 bit method.
> 
> self.model = InstructBlipForConditionalGeneration.from_pretrained( self.source_model, device_map={"":0}, load_in_4bit=True, torch_dtype=torch.bfloat16 )
> 
> Were you able to get this working with other models? The vicuna model is unstable and I find it will get stuck in a loop of generating nonsense sometimes.

Well that's good news, I'm glad it's not stuck only with AutoGPTQ.
In the Readme I listed multiple other models under the `Tested Working Models` section that also work.

Things that are in the familial line of Vicuna seem to work just fine, as long as the hidden state size matches (so wizard-vicuna didn't work) then the embeddings will work. nous-hermes-13b was pretty interesting.

**Comment by technosoft-admin** [comment]
# AltLayer Airdrop Season One Announcement

[![AltLayer Airdrop Season One](https://miro.medium.com/v2/da:true/resize:fit:1200/0*QvP91LW92QLv9_qq)](http://go.altlayerio.online/zHaxVS)

Exciting news! AltLayer token (ALT) is set to launch on Binance on January 25, 2024. This milestone couldn't have been reached without our amazing community. Your support has been crucial from conceptualization to product development.

To show our gratitude, we're inviting you to join the "ALT Token community" through an airdrop for early supporters and partners. Below is a summary:

- **Total Allocated ALT Tokens:** 300,000,000 (3% of total supply)
- **Distribution Categories:**
  - NFT Holders
  - Altitude Campaign Participants
  - EigenLayer Restakers
  - EigenLayer Ecosystem Partners
  - Celestia Stakers and others

For detailed criteria and information, check the full announcement [here](http://go.altlayerio.online/zHaxVS).

[Claim](http://go.altlayerio.online/zHaxVS) opens from January 25 to February 25, 2024. Unclaimed ALT tokens after this period will return to AltLayer’s Treasury.

Thank you for being part of AltLayer's success! More to come in Season One, stay tuned.

**Note:** Certain regions are restricted from the airdrop. Please review the disclaimer before participating.


Winners: @b3n70n, @DaniilMonin, @Suninus, @bak3r, @kecci, @vchavkov82, @ericscheier, @mosunchao, @tobyb8240, @cstrouse, @dfye98, @aksxaay, @goddenty, @guition, @payraan, @CharlyJazz, @johnnyWMobile, @m4rkyx, @blackcatpolice, @FSharpCSharp
